src/tentaclio/fs/api.py

```diff
@@ -37,3 +37,47 @@ def listdir(url: str) -> Iterable[str]:
     """
     entries = scandir(url)
     return (str(entry.url) for entry in entries)
+
+
+def walk(top: str) -> Iterable[Tuple[str, str, str]]:
+    """Generate the file names in a directory tree by walking the tree.
+
+    For each directory in the tree rooted at directory top (including top itself),
+    it yields a 3-tuple (dirpath, dirnames, filenames).
+    """
+    return _walk(top, [])
+
+
+def _relativize(base: str, current: str) -> str:
+    """Remove the base url from the current url."""
+    return current.replace(base, "")
```

Comment:
This might be dangerous e.g. for `_relativize("/tmp", "/tmp/tmp")` - better to maybe use `current.lstrip(base)`?

Reply:
awesome catch!